com.github.Qv2ray.json

```diff
@@ -0,0 +1,89 @@
+{
+    "app-id": "com.github.Qv2ray",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.13",
+    "sdk": "org.kde.Sdk",
+    "command": "qv2ray",
+    "rename-desktop-file": "qv2ray.desktop",
+    "rename-icon": "qv2ray",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
+        "--socket=pulseaudio",
+        "--filesystem=host",
+        "--talk-name=org.kde.StatusNotifierWatcher",
+        "--talk-name=org.freedesktop.Notifications",
+        "--talk-name=com.canonical.Unity",
+        "--talk-name=org.freedesktop.portal.Fcitx",
+        "--device=dri"
+    ],
+    "modules": [
+        {
+            "name": "grpc",
+            "no-autogen": true,
+            "make-install-args": [
+                "prefix=/app"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/grpc/grpc.git",
+                    "tag": "v1.26.0",
+                    "commit": "de893acb6aef88484a427e64b96727e4926fdcfd"
+                },
+                {
+                    "type": "patch",
+                    "path": "hug-grpc-deadlock.patch"
+                }
+            ]
+        },
+        {
+            "name": "protobuf",
+            "build-options": {
+                "prefix": "/app"
+            },
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/protocolbuffers/protobuf.git",
+                    "tag": "v3.11.2",
+                    "commit": "fe1790ca0df67173702f70d5646b82f48f412b99"
+                }
+            ]
+        },
+        {
+            "name": "qv2ray",
+            "buildsystem": "qmake",
+            "config-opts": [
+                "CONFIG+=no_increase_build_number"
+            ],
+            "build-commands": [
+                "install -Dm644 -t /app/share/metainfo com.github.Qv2ray.appdata.xml"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/lhy0403/Qv2ray.git",
+                    "tag": "v1.99.4",
+                    "commit": "de88bfc69e50bf7c4ce034756720bf06df42612a"
+                },
+                {
+                    "type": "patch",
+                    "path": "qv2ray.patch"
+                },
+                {
+                    "type": "file",
+                    "path": "com.github.Qv2ray.appdata.xml"
```

Comment:
New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html